lib/model/emoji.dart

```diff
@@ -30,7 +30,14 @@ class UnicodeEmojiDisplay extends EmojiDisplay {
   /// The actual Unicode text representing this emoji; for example, "🙂".
   final String emojiUnicode;
 
-  UnicodeEmojiDisplay({required super.emojiName, required this.emojiUnicode});
+  /// The Zulip "emoji code" for this emoji; for example, "1f642" for "🙂" emoji.
+  final String emojiCode;
```

Comment:
Ideally this doesn't belong here — this class is meant to be strictly the information used for displaying the emoji.

What are the places that would need to change without this field? I suspect they can pass around an EmojiCandidate instead.